### ISSUE TITLE
code for search functionality to select specific text in jquery

### ISSUE THREAD

**Issue by HDBFS** [HDBFS/URLS#1]
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<asp:Content ID="Content1" ContentPlaceHolderID="Head" runat="server">
</asp:Content>
<asp:Content ID="Content2" ContentPlaceHolderID="ContentPlaceHolder1" runat="server">
    
    <script src="../Scripts/jquery-1.10.2.js"></script>
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>
    <script src="https://code.jquery.com/jquery-3.1.1.js"></script>
    <script src="https://code.jquery.com/jquery-3.1.1.min.js"></script>
    <script type="text/javascript">

    function removeHighlighting(highlightedElements) {
        debugger;
        highlightedElements.each(function () {
            var element = $(this);
            element.replaceWith(element.html());
        })
    }

    function addHighlighting(element, textToHighlight) {
        debugger;
        var text = element.text();
        var highlightedText = '<em>' + textToHighlight + '</em>';
        var newText = text.replace(textToHighlight, highlightedText);

        element.html(newText);
    }
    $(document).ready(function () {
        debugger;
        loadTabletovariable();
    });
    var table = "";
    function loadTabletovariable()
    {
        debugger;
        table = $("table")[0].innerHTML;
    }
    function myFunction1() {
        debugger;

        var input, filter;
        // var value = $(this).val();
        input = document.getElementById("search");
        value = input.value;
        $("table")[0].innerHTML = "";
        $("table")[0].innerHTML = table;
        if (value == "") return;
        $("table tr").each(function (index) {
            debugger;
                $row = $(this);
                $row.show();
                var $tdElement = $row.find("a,p");
                //var id = $tdElement[0].innerHTML;
                var isExists = false;
                $tdElement.each(function (i, obj) {
                    debugger;
                    if (highlight($(this)[0], $(this)[0].innerHTML) <= 0) {
                        isExists = true;
                    }

                });
                if(!isExists)
                    $row.hide();

                //if (matchedIndex != 0) {
                //    $row.hide();
                //}
                //else {
                //    //addHighlighting($tdElement, value);
                //    highlight(id);
                //    $row.show();
                //}
        });
    };

    function highlight(Element, text) {
        debugger;
        var innerHTML = text;
        var index = innerHTML.indexOf(value);
        if (index >= 0) {
            innerHTML = innerHTML.substring(0, index) + "<span class='highlight'>" + innerHTML.substring(index, index + value.length) + "</span>" + innerHTML.substring(index + value.length);
            Element.innerHTML = innerHTML;
            return 0;
        }
        else
            return 1;
    }
    </script>
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width,initial-scale=1.0">
    <style type="text/css">
            table, tr, td, th {
                border: 1px solid blue;
                padding: 2px;
            }

                table th {
                    background-color: #999999;
                }

            em {
                background-color: yellow;
            }

            .red {
                color: red;
            }

            .highlight {
                background-color: yellow;
            }
    </style>

    <div class="container" style="width: auto; max-height: 750px; overflow-y: scroll">
               <span class="glyphicon glyphicon-search"></span>

        <div class="panel-group" id="accordion">

            <table id="myTable" class="">
                <tr id="tr1">
                    <td class="tdclass">
                        <div class="panel panel-default" id="one">
                            <div class="panel-heading">
                                <span class="glyphicon glyphicon-hand-right"></span>&nbsp<a data- toggle="collapse" data-parent="#accordion" href="#collapse1" style="font-size: 12px">Where can I get e-Pehchan Card ?</a>
                            </div>
                            <div id="collapse1" class="panel-collapse collapse in">
                                <div class="panel-body">
                                    <p id="pgid" style="font-size: 11px; font-weight: normal; font-family: Tahoma, Arial, Helvetica, Sans-Serif; padding-left: 5px; padding-right: 10px; padding-bottom: 8px; text-align: justify;">
                                        Employee is requested to approach his HR/BM for e-Pehchan Card.
                                    </p>
                                </div>
                            </div>
                        </div>
                    </td>
                </tr>
                <tr id="tr2">
                    <td class="tdclass">
                        <div class="panel panel-default">
                            <div class="panel-heading">
                                <span class="glyphicon glyphicon-hand-right"></span>&nbsp<a data-toggle="collapse" data-parent="#accordion" href="#collapse2" style="font-size: 12px">What is to be done with e-Pehchan Card ?</a>
                            </div>
                            <div id="" class="">
                                <div class="panel-body">
                                    <p style="font-size: 11px; font-weight: normal; font-family: Tahoma, Arial, Helvetica, Sans-Serif; padding-left: 5px; padding-right: 10px; padding-bottom: 8px; text-align: justify; color: black;">
                                        Get the e-Pehchan Card signed by HR BM & visit to the nearest ESIC Dispensary register yourself to avail various medical benefit from ESIC.
                                    </p>
                                </div>
                            </div>
                        </div>
                    </td>
                </tr>
            </table>          
        </div>
        <input type="text" id="search" onkeyup="myFunction1()" placeholder=" live search" />

--------new code--------------------
<script src="../Scripts/jquery-1.10.2.js"></script>
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>
<script src="https://code.jquery.com/jquery-3.1.1.js"></script>
<script src="https://code.jquery.com/jquery-3.1.1.min.js"></script>
<script type="text/javascript">
    //function removeHighlighting(highlightedElements) {
    //    debugger;
    //    highlightedElements.each(function () {
    //        var element = $(this);
    //        element.replaceWith(element.html());
    //    })
    //}

    //function addHighlighting(element, textToHighlight) {
    //    var text = element;
    //    var highlightedText = '<em>' + textToHighlight + '</em>';
    //    var newText = text.replace(textToHighlight, highlightedText);

    //    element.html(newText);
    //}

    //function myFunction1() {
    //    var iptxt = '';
    //    iptxt = document.getElementById("myInput").value;
    //    if (iptxt != "" && iptxt != null && iptxt != undefined) {
    //        var context = $("#pgid").text();//text();
    //        //$("divid:contains('" + iptxt + "')").each(function () {
    //        //    var regex = new RegExp(iptxt, 'gi');
    //        //    $(this).html($(this).text().replace(regex, "<span class='red'>" + iptxt + "</span>"));
    //        //});
    //        $("#pgid").html(context.replace(iptxt, '<span class="red">' + iptxt + '</span>'));
    //    }
    //}

    //function myFunction2() {
    //    var input, filter, table, tr, td, i;
    //    input = document.getElementById("myInput");
    //    filter = input.value.toUpperCase();
    //    table = document.getElementById("myTable");
    //    tr = table.getElementsByTagName("tr");
    //    for (i = 0; i < tr.length; i++) {

    //    }
    //}

    //function myFunction() {
    //    var input, filter, table, tr, td, i;
    //    input = document.getElementById("myInput");
    //    filter = input.value.toUpperCase();
    //    table = document.getElementById("myTable");
    //    tr = table.getElementsByTagName("tr");
    //    for (i = 0; i < tr.length; i++) {
    //        td = tr[i].getElementsByTagName("td")[0];
    //        if (td) {
    //            debugger;
    //            //if (td.innerHTML.toUpperCase().indexOf(filter) > -1) {
    //            if (td.innerText.toUpperCase().indexOf(filter) > -1) {
    //                var value = td.innerHTML.toUpperCase().indexOf(filter);
    //                debugger;
    //                if (filter !== "") {
    //                    $row = $(this);
    //                    var $tdElement = $row.find("td:first");
    //                    var id = $tdElement.text();
    //                    var matchedIndex = id.indexOf(value);
    //                    //if (matchedIndex != 0) {
    //                    //    $row.hide();
    //                    //}
    //                    //else {
    //                    //addHighlighting(filter, value);
    //                    //  $row.show();
    //                    //}

    //                    //text = $(this).text().replace(/\s+/g, ' ');
    //                    //tr[i].style.display = "";
    //                    //$("tr[i] > td .tdclass").find(filter).css("color", "Green");

    //                    //$('.pclass').css('color', 'red');
    //                    if (i == 0) {                            
    //                        //var context = $("#pgid").text();
    //                        var context = td.innerText;//$("#pgid").text();
    //                        //$(this).html(context.replace(filter, '<span class="red">' + filter + '</span>'));
    //                        $(tr[i].getElementsByTagName("td")[0]).html(context.replace(filter, "<span style='color:red'>" + filter + "</span>"));
    //                        $("#tr2").hide();
    //                        $("#tr1").show();
    //                    }
    //                    else if (i == 1) {                            
    //                        var context = td.innerText;
    //                        $(tr[i].getElementsByTagName("td")[0]).html(context.replace(filter, "<span class='red'>" + filter + "</span>"));
    //                        $("#tr1").hide();
    //                        $("#tr2").show();
    //                    }
    //                    //$('tr[i] > td.pclass').each(function () {
    //                    //    if ($(this).text() == filter) {
    //                    //        $(this).style('color', 'red');
    //                    //    } // similarly for other statuses
    //                    //});

    //                    // $("tr[i] > td .").find(filter).css("color", "Green");
    //                    //alert($("tr[i] > td .tdclass").text());
    //                    //$("tr[i] td div p:contains(filter)").css("background-color", "Green");

    //                    //$('tr[i] td .p1').find(filter).css('background-color', 'red');

    //                    //$('.p1').find(filter).css('background-color', 'green');
    //                }
    //            }

    //            else {
    //                debugger;
    //                //tr[i].style.display = "none";
    //            }
    //        }
    //    }
    //}

    function removeHighlighting(highlightedElements) {
        debugger;
        highlightedElements.each(function () {
            var element = $(this);
            element.replaceWith(element.html());
        })
    }

    function addHighlighting(element, textToHighlight) {
        debugger;
        var text = element.text();
        var highlightedText = '<em>' + textToHighlight + '</em>';
        var newText = text.replace(textToHighlight, highlightedText);

        element.html(newText);
    }
    $(document).ready(function () {
        loadTabletovariable();
    });
    var table = "";
    function loadTabletovariable()
    {
        debugger;
        table = $("table")[0].innerHTML;
    }
    function myFunction1() {
        debugger;
        
        var input, filter;
        // var value = $(this).val();
        input = document.getElementById("search");
        value = input.value;
        $("table")[0].innerHTML = "";
        $("table")[0].innerHTML = table;
        if (value == "") return; 
        $("table tr").each(function (index) {
            debugger;
                $row = $(this);
                $row.show();
                var $tdElement = $row.find("a,p");
                //var id = $tdElement[0].innerHTML;
                var isExists = false;
                $tdElement.each(function (i, obj) {
                    debugger;
                    if (highlight($(this)[0], $(this)[0].innerHTML) <= 0) {
                        isExists = true;
                    }

                });
                if(!isExists)
                    $row.hide();
                

                //if (matchedIndex != 0) {
                //    $row.hide();
                //}
                //else {
                //    //addHighlighting($tdElement, value);
                //    highlight(id);
                //    $row.show();
                //}
        });
    };

    function highlight(Element, text) {
        debugger;
        var innerHTML = text;
        var index = innerHTML.indexOf(value);
        if (index >= 0) {
            innerHTML = innerHTML.substring(0, index) + "<span class='highlight'>" + innerHTML.substring(index, index + value.length) + "</span>" + innerHTML.substring(index + value.length);
            Element.innerHTML = innerHTML;
            return 0;
        }
        else
            return 1;
    }
</script>
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width,initial-scale=1.0">
<style type="text/css">
    table, tr, td, th {
        border: 1px solid blue;
        padding: 2px;
    }

        table th {
            background-color: #999999;
        }

    em {
        background-color: yellow;
    }

    .red {
        color: red;
    }

    /** {
        box-sizing: border-box;
    }

    #myInput {
        background-image: url('/css/searchicon.png');
        background-position: 10px 10px;
        background-repeat: no-repeat;
        width: 100%;
        font-size: 16px;
        padding: 12px 20px 12px 40px;
        border: 1px solid #ddd;
        margin-bottom: 12px;
    }

    #myTable {
        border-collapse: collapse;
        width: 100%;
        border: 1px solid #ddd;
        font-size: 18px;
    }

        #myTable th, #myTable td {
            text-align: left;
            padding: 12px;
        }

        #myTable tr {
            border-bottom: 1px solid #ddd;
        }

            #myTable tr.header, #myTable tr:hover {
                background-color: #f1f1f1;
            }

    .highlight {
        background-color: yellow;
    }

    .ScrollStyle {
        max-height: 150px;
        overflow-y: scroll;
    }*/
    .highlight {
  background-color: yellow;
}
</style>

<div class="container" style="width: auto; max-height: 750px; overflow-y: scroll">
    <input type="text" id="myInput" onkeyup="myFunction()" placeholder="Search for Questions & Answers.."
        title="Type in a name" style="width: 700px" />
    <span class="glyphicon glyphicon-search"></span>

    <div class="panel-group" id="accordion">

        <%--<table id="myTable" class="">
            <tr id="tr1">
                <td class="tdclass">
                    <div class="panel panel-default" id="one">
                        <div class="panel-heading">
                            <span class="glyphicon glyphicon-hand-right"></span>&nbsp<a data- toggle="collapse" data-parent="#accordion" href="#collapse1" style="font-size: 12px">Where can I get e-Pehchan Card ?</a>
                        </div>
                        <div id="collapse1" class="panel-collapse collapse in">
                            <div class="panel-body">
                                <p id="pgid" style="font-size: 11px; font-weight: normal; font-family: Tahoma, Arial, Helvetica, Sans-Serif; padding-left: 5px; padding-right: 10px; padding-bottom: 8px; text-align: justify;">
                                    Employee is requested to approach his HR/BM for e-Pehchan Card.
                                </p>
                            </div>
                        </div>
                    </div>
                </td>
            </tr>
            <tr id="tr2">
                <td class="tdclass">
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <span class="glyphicon glyphicon-hand-right"></span>&nbsp<a data-toggle="collapse" data-parent="#accordion" href="#collapse2" style="font-size: 12px">What is to be done with e-Pehchan Card ?</a>
                        </div>
                        <div id="" class="">
                            <div class="panel-body">
                                <p style="font-size: 11px; font-weight: normal; font-family: Tahoma, Arial, Helvetica, Sans-Serif; padding-left: 5px; padding-right: 10px; padding-bottom: 8px; text-align: justify; color: black;">
                                    Get the e-Pehchan Card signed by HR BM & visit to the nearest ESIC Dispensary register yourself to avail various medical benefit from ESIC.
                                </p>
                            </div>
                        </div>
                    </div>
                </td>
            </tr>
        </table>--%>

        
    <div class="panel-group">

        <table>
            <tr>
                <td>
                    <div class="panel panel-default" id="one">
                        <div class="panel-heading">
                            <span class="glyphicon glyphicon-hand-right"></span>&nbsp<a href="#collapse1" style="font-size: 12px">Where can I get e-

Pehchan Card ?</a>
                        </div>
                        <div>
                            <div class="panel-body">
                                <p style="font-size: 11px; font-weight: normal; font-family: Tahoma, Arial, 

Helvetica, Sans-Serif; padding-left: 5px; padding-right: 10px; padding-bottom: 8px; text-align: justify; 

color: black;" >
                                   Employee is requested to approach his HR/BM for e-Pehchan Card.
                                </p>
                            </div>
                        </div>
                    </div>
                </td>
            </tr>
            <tr>
                <td>
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <span class="glyphicon glyphicon-hand-right"></span>&nbsp<a href="#collapse2" style="font-size: 12px">What is to be done with 

e-Pehchan Card ?</a>
                        </div>
                        <div id="collapse2" class="panel-collapse collapse">
                            <div class="panel-body">
                                <p style="font-size: 11px; font-weight: normal; font-family: Tahoma, Arial, 

Helvetica, Sans-Serif; padding-left: 5px; padding-right: 10px; padding-bottom: 8px; text-align: justify; 

color: black;">
                                    Get the e-Pehchan Card signed by HR/BM & visit to the nearest ESIC 

Dispensary register yourself to avail various medical benefit from ESIC.
                                </p>
                            </div>
                        </div>
                    </div>
                </td>
            </tr>

</table>
<br />
<input type="text" id="search" onkeyup="myFunction1()" placeholder="live search" />

    </div>
</div>
